blocks/color-palette/index.js

```diff
@@ -17,94 +16,62 @@ import { __, sprintf } from '@wordpress/i18n';
-		this.closeOnClickOutside = this.closeOnClickOutside.bind( this );
-		this.bindToggleNode = this.bindToggleNode.bind( this );
-	}
+function ColorPalette( { colors, value, onChange } ) {
+	return (
+		<div className="blocks-color-palette">
+			{ colors.map( ( color ) => {
+				const style = { color: color };
+				const className = classnames( 'blocks-color-palette__item', { 'is-active': value === color } );
 
-	togglePicker() {
-		this.setState( ( state ) => ( { opened: ! state.opened } ) );
-	}
-
-	closeOnClickOutside( event ) {
-		const { opened } = this.state;
-		if ( opened && ! this.toggleNode.contains( event.target ) ) {
-			this.togglePicker();
-		}
-	}
-
-	bindToggleNode( node ) {
-		this.toggleNode = node;
-	}
-
-	render() {
-		const { colors, value, onChange } = this.props;
-		return (
-			<div className="blocks-color-palette">
-				{ colors.map( ( color ) => {
-					const style = { color: color };
-					const className = classnames( 'blocks-color-palette__item', { 'is-active': value === color } );
-
-					return (
-						<div key={ color } className="blocks-color-palette__item-wrapper">
-							<button
-								type="button"
-								className={ className }
-								style={ style }
-								onClick={ () => onChange( value === color ? undefined : color ) }
-								aria-label={ sprintf( __( 'Color: %s' ), color ) }
-								aria-pressed={ value === color }
-							/>
-						</div>
-					);
-				} ) }
+				return (
+					<div key={ color } className="blocks-color-palette__item-wrapper">
+						<button
+							type="button"
+							className={ className }
+							style={ style }
+							onClick={ () => onChange( value === color ? undefined : color ) }
+							aria-label={ sprintf( __( 'Color: %s' ), color ) }
+							aria-pressed={ value === color }
+						/>
+					</div>
+				);
+			} ) }
 
-				<div className="blocks-color-palette__item-wrapper blocks-color-palette__custom-color">
+			<Dropdown
+				className="blocks-color-palette__item-wrapper blocks-color-palette__custom-color"
+				contentClassName="blocks-color-palette__picker "
+				renderToggle={ ( { isOpen, onToggle } ) => (
 					<button
 						type="button"
-						aria-expanded={ this.state.opened }
+						aria-expanded={ isOpen }
 						className="blocks-color-palette__item"
-						onClick={ this.togglePicker }
-						ref={ this.bindToggleNode }
+						onClick={ onToggle }
 						aria-label={ __( 'Custom color picker' ) }
 					>
 						<span className="blocks-color-palette__custom-color-gradient" />
 					</button>
-					<Popover
-						isOpen={ this.state.opened }
-						onClickOutside={ this.closeOnClickOutside }
-						className="blocks-color-palette__picker"
-					>
-						<ChromePicker
-							color={ value }
-							onChangeComplete={ ( color ) => {
-								onChange( color.hex );
-								this.togglePicker();
-							} }
-							style={ { width: '100%' } }
-							disableAlpha
-						/>
-					</Popover>
-				</div>
+				) }
+				renderContent={ () => (
```

Comment:
Did you consider just passing this as children? Even if it were a function callback?

Reply:
Yes, I did, it was my first option, but then I hesitated whether the `renderToggle` should be the children function because it's the one being displayed as a child and not in a portal. 

The answer was not obvious to me, which one to choose so I preferred avoiding it completely and being consistent between `renderToggle` and `renderContent`